breakpoint.c

```diff
@@ -0,0 +1,68 @@
+#include "breakpoint.h"
+#include <assert.h>
+
+static inline int cmp(const void *arg0, const void *arg1)
+{
+    riscv_word_t *a = (riscv_word_t *) arg0, *b = (riscv_word_t *) arg1;
+    return (*a < *b) ? _CMP_LESS : (*a > *b) ? _CMP_GREATER : _CMP_EQUAL;
+}
+
+void breakpoint_map_init(breakpoint_map_t *breakpoint_map)
+{
+    assert(breakpoint_map);
+
+    breakpoint_map->map = map_init(riscv_word_t, breakpoint_t, cmp);
+}
+
+bool breakpoint_map_insert(breakpoint_map_t *breakpoint_map, riscv_word_t addr)
+{
+    assert(breakpoint_map);
+
+    breakpoint_t breakpoint = (breakpoint_t) {.addr = addr, .orig_instr = 0};
```

Comment:
Simply name the variable in shorter fashion. i.e., something like `bp` or `bkpt`.